### PR TITLE
Draw & plot multiple points on click

### PR DESCRIPTION
**Pull request by prodyut1978**
## Background
This PR is the continuation of the previous Jjyy igoss #970. The client requested to select multiple points on click in an area of their interest to generate the JJYY file output.

## Why did you take this approach?
The addition of this new feature can help the end user to select multiple points on click and can plot them together in the same plot window. 

## Anything in particular that should be highlighted?
1. The user needs to select multiple points and then hit the plot button option to generate plot. The hitting plot button will end the end of point selection. To start new point selection, the user needs to Reset Map option. 

## Screenshot(s)

Capture1

## Checks
- [ ] I ran unit tests.
- [y ] I've tested the relevant changes from a user POV.
- [ ] I've formatted my Python code using `black .`.

_Hint_ To run all python unit tests run the following command from the root repository directory:
```sh
bash run_python_tests.sh
```

### CODE REVIEW

File: oceannavigator/frontend/src/components/OceanNavigator.jsx [hunk 1]
```diff
@@ -134,8 +135,10 @@ class OceanNavigator extends React.Component {
   }
 
   // Turns off map drawing
-  removeMapInteraction(mode) {
-    this.mapComponent.removeMapInteractions(mode);
+  removeMapInteraction(mode="All") {
```
Comment (htmlboss) [inline oceannavigator/frontend/src/components/OceanNavigator.jsx:138]:
```suggestion
  removeMapInteraction(mode = "All") {
```

Comment (htmlboss) [reply oceannavigator/frontend/src/components/OceanNavigator.jsx:138]: Can `mode` be renamed to be something more descriptive? I don't understand what it means.

File: oceannavigator/frontend/src/components/OceanNavigator.jsx [hunk 2]
```diff
@@ -581,7 +579,7 @@ class OceanNavigator extends React.Component {
         modalTitle = "";
         break;
     }
-    if (this.state.names && this.state.names.length > 0) {
+    if (this.state.modal !="point" && this.state.names && this.state.names.length > 0) {
```
Comment (htmlboss) [inline oceannavigator/frontend/src/components/OceanNavigator.jsx:582]:
```suggestion
    if (this.state.modal !== "point" && this.state.names && this.state.names.length > 0) {
```

File: oceannavigator/frontend/src/components/MapToolbar.jsx [hunk 3]
```diff
@@ -72,6 +72,7 @@ class MapToolbar extends React.Component {
       name = elem.name;
     }
     this.props.action(name);
+    this.props.disablePlotInteraction();
```
Comment (htmlboss) [inline oceannavigator/frontend/src/components/MapToolbar.jsx:75]: This should be added to the props validation at the bottom of this file.

File: oceannavigator/frontend/src/components/Map.jsx [hunk 4]
```diff
@@ -1260,17 +1266,18 @@ export default class Map extends React.PureComponent {
       // Disable zooming when drawing
       this.controlDoubleClickZoom(false);
       const lonlat = olproj.transform(e.feature.getGeometry().getCoordinates(), this.props.state.projection, "EPSG:4326");
+      let lonlat_vector = [...this.state.lonlat, [lonlat[1], lonlat[0]]]
```
Comment (htmlboss) [inline oceannavigator/frontend/src/components/Map.jsx:1269]: If a variable is not being modified, always declare it `const`.
```suggestion
      const drawn_lonlats = [...this.state.lonlat, [lonlat[1], lonlat[0]] ];
```
FYI JS does not have a vector concept; only arrays.

File: oceannavigator/frontend/src/components/Map.jsx [hunk 5]
```diff
@@ -1260,17 +1266,18 @@ export default class Map extends React.PureComponent {
       // Disable zooming when drawing
       this.controlDoubleClickZoom(false);
       const lonlat = olproj.transform(e.feature.getGeometry().getCoordinates(), this.props.state.projection, "EPSG:4326");
```
Comment (htmlboss) [inline oceannavigator/frontend/src/components/Map.jsx:1268]: I think can avoid all of the messy `[lonlat[1], lonlat[0]]` array allocations by simply reversing this array. Note you'll have to change your variable names from `lonlat` to `latlon` where appropriate.

```js
const latlon = olproj
    .transform(e.feature.getGeometry().getCoordinates(), this.props.state.projection, "EPSG:4326")
    .reverse();

const drawn_latlons = [...this.state.latlon, latlon];
```

**Review by htmlboss — COMMENTED** [review]
left some comments that may help with simplifying the code a bit

File: oceannavigator/frontend/src/components/Map.jsx [hunk 6]
```diff
@@ -1259,18 +1266,21 @@ export default class Map extends React.PureComponent {
     draw.on("drawend", function (e) {
       // Disable zooming when drawing
       this.controlDoubleClickZoom(false);
-      const lonlat = olproj.transform(e.feature.getGeometry().getCoordinates(), this.props.state.projection, "EPSG:4326");
+      const latlon = olproj
+      .transform(e.feature.getGeometry().getCoordinates(), this.props.state.projection, "EPSG:4326")
+      .reverse(); 
+      const drawn_latlons = [...this.state.latlon, latlon];
```
Comment (JustinElms) [inline oceannavigator/frontend/src/components/Map.jsx:1272]:
``` suggestion 
      const latlon = olproj
            .transform(e.feature.getGeometry().getCoordinates(), this.props.state.projection, "EPSG:4326")
            .reverse(); 

      const drawn_latlons = [...this.state.latlon, latlon];
```

This should help make this section a little easier to read.

**Review by JustinElms — APPROVED** [review]

**Pull request merged**